### DISCUSSION TITLE
access to DS18x20 sensor in scripts does not work

### DISCUSSION THREAD

**Discussion by dad401** [arendst/Tasmota#13516] [Q&A] [answered]
I try to access the temperature of a DS18B20 sensor in Tasmota script, but somehow it's not working. 

part of my script:
```
>T
t1=BME280#Temperature
t2=DS1820_1#Temperature
i2=DS1820_1#Id
```

I got the temperature for BME280 but not for DS18B20. I also tried "DS18B20-1" and "DS18B20_1". t2 and i2 keep their default values (initialized at >D)

console log:
`11:25:00.880 RSL: SENSOR = {"Time":"2021-10-30T11:25:00","ANALOG":{"Temperature":204.9},"ENERGY":{"TotalStartTime":"2021-09-23T20:30:04","Total":0.000,"Yesterday":0.000,"Today":0.000,"Period": 0,"Power": 0,"ApparentPower": 0,"ReactivePower": 0,"Factor":0.00,"Voltage": 0,"Current":0.000,"ResetTotalActive":0.000,"MaxPower":0,"PhaseAngle":0.00},"DS18B20-1":{"Id":"0300A2799076","Temperature":21.2},"DS18B20-2":{"Id":"0300A2799D5E","Temperature":21.0},"BME280":{"Temperature":20.9,"Humidity":100.0,"DewPoint":20.9,"Pressure":955.0},"PressureUnit":"hPa","TempUnit":"C"}`

What I'm missing here?

**Comment by barbudor** [comment, marked as answer]
I see your sensor message is using '-' instead of '_'
You should use 'setoption64 1' to enforce underscore instead of hyphen

**Comment by dad401** [comment]
Thanks! Setting option64 to 1 - and changed the script as follows - helped:
```
>D
t1=0
t2=0
i2=""
DT="DisplayText"
tunit="C"

>B
=>%DT% [z]

>T
t1=BME280#Temperature
t2=DS18B20_1#Temperature
i2=DS18B20_1#Id


>S
// update display every teleperiod time
if upsecs%tper==0
then
dprec2
->%DT% [f1p7x0y5]%t1% %tunit%
->%DT% [f1p7x0y20]%t2% %tunit%
->%DT% [f1p7x0y35]%i2%


; now update screen
->%DT% [d]
endif
```
But I'm still wondering why `t2=DS18B20-1#Temperature` didn't work?!?! Some mystery...

**Comment by dad401** [comment]
Missed this explanation: "Remark: json variable names (like all others) may not contain math operators like - , you should set SetOption64 1 to replace - (dash) with _ (underscore). Zigbee sensors will not report to this section, use E instead."

No mystery any more ;-)

Solved!
